File: Day23/lenOfLongIncSubArr.cpp
#include <iostream>
using namespace std;

int lenOfLongIncSubArr(int arr[], int n)
{
    int maxLength = 0;
    int length = 0;

    for (int i = 0; i < n; i++)
    {
        if (arr[i] < arr[i + 1])
        {
            length++;
        }
        else
        {
            maxLength = max(length + 1, maxLength);
            length = 0;
        }
    }

    return maxLength;
}
int main()
{
    int arr[] = {5, 6, 3, 5, 7, 8, 9, 1, 2};
    int n = sizeof(arr) / sizeof(arr[0]);
    std::cout << lenOfLongIncSubArr(arr, n) << std::endl;

    int arr1[] = {12, 13, 1, 5, 4, 7, 8, 10, 10, 11};
    int m = sizeof(arr1) / sizeof(arr1[0]);
    std::cout << lenOfLongIncSubArr(arr1, m) << std::endl;
}

File: Day23/lenOfLongIncSubArr.js
function lenOfLongIncSubArr(arr) {
    let length = 0;
    let max = 0;
    let start = 0;
    let res = []; // want to see the data in subarray

    for (let i = 0; i < arr.length - 1; i++) {
        if (arr[i] < arr[i + 1]) {
            length++;
        } else {
            if (length > max) {
                res = [];
                res.push(...arr.slice(start, i + 1));
                max = length;
            }
            length = 0;
            start = i + 1;
        }
    }


    return res.length;
}

let a = [5, 6, 3, 5, 7, 8, 9, 1, 2];
console.log(lenOfLongIncSubArr(a));
a = [];
console.log(lenOfLongIncSubArr(a));